src/CloudFoundry.WinDEA/Agent.cs

```diff
@@ -2429,7 +2431,21 @@ private void TheReaper()
 
                     if (instance.Properties.Stopped && !instance.Properties.StagingDone)
                     {
-                        this.AfterStagingFinished(instance);
+                        if (!instance.StopRequested)
+                        {
+                            this.AfterStagingFinished(instance);
+                        }
+                        else
+                        {
+                            Logger.Info("Destroying prison for staging instance {0}", instance.Properties.TaskId);
```

Comment:
Would it be better to only execute this branch if instance.CompileProcess != null ?